### PR TITLE
smaq2oss: pairs and banks raw data, decodes nothing yet

### PR DESCRIPTION
**Pull request by abdulsaheel**
pairs-only adapter for the SMA-Q2-OSS watch. no signals declared, no commands sent.

## Summary by Sourcery

Add experimental SMA-Q2-OSS support that pairs with the watch and archives its notifications as un decoded raw frames.

New Features:
- Add experimental SMA-Q2-OSS pairing, background synchronization, and manual sync support.
- Archive incoming SMA-Q2-OSS notifications as raw data without decoding signals or issuing commands.

Enhancements:
- Register the SMA-Q2-OSS service and characteristics, expose it in device selection and profile UI, and enforce serialized, bounded listen sessions with cleanup.

Tests:
- Add registry and adapter tests covering the empty signal set, raw frame archiving, ignored empty notifications, absence of writes, and absence of offload checkpoints.

## Summary by CodeRabbit

* **New Features**
  * Added pairing and synchronization support for SMA-Q2-OSS smartwatches.
  * Incoming watch data is stored for later access.
  * Added device identification and smartwatch iconography in the device picker.
  * Added manual and background synchronization, including sync status feedback.

### CODE REVIEW

**Comment by abdulsaheel** [comment]
@coderabbitai review

**Comment by abdulsaheel** [comment]
@coderabbitai review

**Comment by abdulsaheel** [comment]
@coderabbitai review

**Comment by abdulsaheel** [comment]
@coderabbitai review

**Comment by abdulsaheel** [comment]
@coderabbitai review

**Comment by abdulsaheel** [comment]
@coderabbitai review

File: lib/ble/smaq2oss_link.dart [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    final deviceId = row['id'] as String?;
+    final remoteId = row['remote_id'] as String?;
+    if (deviceId == null || remoteId == null || remoteId.isEmpty) return false;
+    if (deviceId == LocalDb.kPrimaryDeviceId) {
+      debugPrint('[smaq2oss] refusing to sync: the row claims the primary '
+          'device id — re-pair it with a minted id.');
+      return false;
+    }
+
+    try {
```
[the remark this answers is not in the document]
Comment (abdulsaheel) [reply lib/ble/smaq2oss_link.dart:70]: fixed in 9c9c19a — pairedRow() and the stored-row field validation are now inside the same try block as the rest of _sync, so a LocalDb.deviceRows() failure or a bad cast converts to false instead of escaping sync()'s never-throws contract. (the ✅ note above cites a4f230a — that commit is the separate stop()-disconnect fix that landed after; this one is 9c9c19a.)
